Docs/deep_linking.md

```diff
@@ -0,0 +1,169 @@
+# URL structure for deeplinking
+
+ This describes the structure of the URL so that it is easy for developers to create deep links from other applications directly to a well/wellbore/object/curve data etc.
+
+## Paramemter encoding:
+
+All server and uid parameters must be URL encoded. 
+
+## Common parameters:
+### objectGroup
+
+This paramater can have values:
+  - BhaRun
+  - ChangeLog
+  - FluidsReport
+  - FormationMarker
+  - Message
+  - MudLog
+  - Log
+  - Rig
+  - Risk
+  - Trajectory
+  - Tubular
+  - WbGeometry
+
+### logType
+
+This paramater can have values:
+  - time
+  - depth
+
+### filterType
+
+This paramater can have values:
+  - Wellbore
+  - Log
+  - Rig
+  - Service%20Company
+
+## Navigation paths:
+
+### Wells View
+/servers/:serverUrl/wells
+
+### Wellbores View
+/servers/:serverUrl/wells/:wellUid/wellbores
+
+### Object Groups
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups
+
+### Objects View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/objects
+
+### Object View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/objects/:objectUid
+
+### Log Types View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/logtypes
+
+### Log Objects View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/logtypes/:logType/objects
+
+### Log Object View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/logtypes/:logType/objects/:objectUid
+
+### Filtering Rows Of Views
+
+Most pages can receive a filter to filter the rows of the table in that page. Those filters are added at the end of the url.
+For example showing only rows where parameter "name" has value "test":
+
+?filter={"name":"test"}
+
+Url encoded version which needs to be used:
+
+%3Ffilter%3D%7B%22name%22%3A%22test%22%7D
+
+
+### Log Curve Values View
+/servers/:serverUrl/wells/:wellUid/wellbores/:wellboreUid/objectgroups/:objectGroup/logtypes/:logType/objects/:objectUid/curvevalues
+
+#### Additional params for searching
+
+If there is a need to search in result, following parameters need to be filled up:
+
+  - startIndex
+  - endIndex
+  - mnemonics
+
+StartIndex and endIndex could be numbers for depth based logs or datetime in ISO format  - "YYYY-MM-DDTHH:mm:ss.sssZ" for time based logs
+
+Samples:
+
+##### Time sample:
+?startIndex=2024-09-06T00:00:00Z&endIndex=2024-09-06T00:01:00Z&mnemonics=["mnem1","mnem2","mnem3","mnem4"]
+
+url encoded version which should be used:
+%3FstartIndex%3D2024-09-06T00%3A00%3A00Z%26endIndex%3D2024-09-06T00%3A01%3A00Z%26mnemonics%3D%5B%22mnem1%22%2C%22mnem2%22%2C%22mnem3%22%2C%22mnem4%22%5D
```

Comment:
Each search param value must be encoded by it's own. `?startIndex=`, `&endIndex=` and `&mnemonics=` should not be encoded.

Reply:
fixed